File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/09 - While Loop Exercise/04. Walking .py
total_steps = 0

while True:
    command = input()
    if command == "Going home":
        steps = int(input())
        total_steps += steps
        break
    steps = int(command)
    total_steps += steps
    if total_steps >= 10000:
        break

if total_steps >= 10000:
    print("Goal reached! Good job!")
else:
    print(f"{10000 - total_steps} more steps to reach goal.")

File: SoftUni Lessons/JavaScript Development/JS Basics May 2023/08. While Loop - Exercise/04. Walking.js
function solution(params) {
    // MasK
    let steps = 0;

    while (params.length > 0) {
        let command = params.shift();

        if (command === 'Going home') {
            steps += Number(params.shift());
            (steps >= 10000) 
            ? console.log(`Goal reached! Good job!\n${steps - 10000} steps over the goal!`) 
            : console.log(`${10000 - steps} more steps to reach goal.`);
            break;
        }
        steps += Number(command);
        if (steps >= 10000) {
            console.log(`Goal reached! Good job!\n${steps - 10000} steps over the goal!`);
            break;
        }
    }
}

solution(["1000",
"1500",
"2000",
"6500"])

// Goal reached! Good job!
// 1000 steps over the goal!

console.log('********************************');

solution(["1500",
"300",
"2500",
"3000",
"Going home",
"200"])

// 2500 more steps to reach goal.
